### PR TITLE
add ctagging discriminators in the default PAT content, small improvement to test script

### PR DESCRIPTION
**Pull request by mverzett**
Just one commit that was scheduled to happen once the AOD RelVals with ctagging were available. Tested with

```
PhysicsTools/PatAlgos/test//runAsyncTests.pl --all
```

and with runTheMatrix, no patological sign.

### CODE REVIEW

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@mverzett: The Jenkins unit test error seems to be caused by your change. Please try to fix the problem.

**Comment by slava77** [comment]
I think the error in the unit tests is valid/relevant: it is triggered by excessive import of b-tagging configuration in PAT: it's trying to re-reconstruct pfCombinedCvsLJetTags instead of reading it from the event.
IMHO, a better choice is to read what's in the AOD file (and cleanup the import/load of configuration in PAT); otherwise you'd need to make sure to redo the TagInfos in the MINIAOD sequence as well.

@gpetruc @arizzi what's the current intended default MINIAOD setup when reading AOD regarding b-tags?
(redo full b-tagging or read from AOD)

**Comment by slava77** [comment]
@mverzett 
if this is really needed in 76X, we better get a solution in the next days; after that the DR campaign in MC will start

**Comment by mverzett** [comment]
@slava77 
Ok, I thought you wanted some feedback from the others first. I will start working on it right away.
Just one comment though.

running the tests with ```./runtests.sh``` and ```cmsRun patTuple_addBTagging_cfg.py``` I can reproduce the error, but running ```./runAsyncTests.pl``` as I was instructed to does not show anything problematic:

```patTuple_addBTagging_cfg.py: done events 1/10, total time 285s, 44 output lines, no exceptions```

I think this should be also fixed, or otherwise removed.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by arizzi** [comment]
@slava77   apologies for seeing this only now, in normal operations I think we want to read from AOD. In cases where a miniAOD campaign is meant to fix/redo something in jets we remake everything. If the ctagging isn't in AOD it is right to trigger what is needed to recompute it.

**Comment by slava77** [comment]
On 11/11/15 2:51 AM, arizzi wrote:
> @slava77 <https://github.com/slava77> apologies for seeing this only
> now, in normal operations I think we want to read from AOD. In cases
> where a miniAOD campaign is meant to fix/redo something in jets we
> remake everything. If the ctagging isn't in AOD it is right to trigger
> what is needed to recompute it.
> 

In this case c-tagging is in AOD.
So, it looks like the default config imports in pat will need to be
cleaned up
from the standard reco btag modules so that the job is not tempted to
re-reconstruct everything.

File: PhysicsTools/PatAlgos/python/tools/jetTools.py [hunk 1]
```diff
@@ -400,7 +400,7 @@ def toolCode(self, process):
                 process.load("RecoBTag.SecondaryVertex.secondaryVertex_cff")
                 process.load("RecoBTag.SoftLepton.softLepton_cff")
                 process.load("RecoBTag.Combined.combinedMVA_cff")
-                process.load("RecoBTag.CTagging.cTagging_cff")
+                process.load("RecoBTag.CTagging.RecoCTagging_cff")
```
Comment (slava77) [inline PhysicsTools/PatAlgos/python/tools/jetTools.py]: it looks like this is the problematic import

Comment (mverzett) [inline PhysicsTools/PatAlgos/python/tools/jetTools.py]: Actually I made that import to get rid of the PATAlgos exception.

**Comment by slava77** [comment]
@mverzett ping

**Comment by arizzi** [comment]
@slava77  btw after yesterday PPD coord discussion we may have a chance to include this in miniaod (@gpetruc)

**Comment by slava77** [comment]
does this mean re-miniaod?

I can see that a reminiaod  in MC can be made about the same time when the data rereco starts (is it Dec ~15?).
I think for this purpose we can do the same as was done for 74X re-miniaod (Giovanni had a rolling PR with all changes needed).

**Comment by davidlange6** [comment]
in any case 80x, validation, then 76x..

**Comment by mverzett** [comment]
Dear All, I got a bit lost. I thought this PR was dead and that was it. What should I do? In which branch?
thank you for the clarifications.

**Comment by slava77** [comment]
@mverzett 
please continue with your 80X PR
This discussion is, effectively,  about possible backports.

**Comment by mverzett** [comment]
Given that #12565 has been merge, how should I/you/we proceed with this one?

**Comment by ferencek** [comment]
@mverzett, as David explained, we wait for a validation in 80X and if no problems are found, this PR can simply be merged as is.

**Comment by pvmulder** [comment]
@ferencek @davidlange6 Given that we have an opportunity to get it in the official miniAOD for the 76X reprocessing, it may be worth to merge it in 76X as soon as possible, avoiding error-prone additional recipes on top of miniAOD. We still have a few days. Is it clear how long the validation of 80X will take? I would just like to understand if there is a chance to get it in 76X in time or if there is no hope at all. 
Also, could you clarify what type of problems to expect when adding a new tagger?

**Comment by ferencek** [comment]
Hi Petra, I'm not exactly sure how long will a 80X validation cycle take and whether it is strictly necessary given the fact that the 80X PR passes all the unit tests. Therefore, I do not expect any problems with the 76X PR. The formal procedure is to merge new features in an open release cycle first and then backport to earlier release cycles and this requirement has been met now.

**Comment by pvmulder** [comment]
@davidlange6 @slava77 do you agree that @mverzett backports the c-tagger to 76X miniAOD now?

**Comment by slava77** [comment]
Petra, does "backport" mean just continuing with this PR?
The 80X version #12565  was merged. So, sure, why not.
It will still have to pass the tests here in 76X.

**Comment by mverzett** [comment]
@slava77 , yes, we mean try to merge this PR. As you can see the last time you guys tested it we were getting DAS errors, definitely not our fault. Moreover, #12565 has **exactly** the same code in it and worked perfectly.

Can you please give it  another try?

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for  https://github.com/cms-sw/cmssw/pull/12288 d6494b0
- it runs in all tests now
- and there are no differences in reco products (or anything monitored in jenkins comparisons)

clean candidate for re-miniAOD

@gpetruc @arizzi

**Comment by mverzett** [comment]
@slava77 whose approval is missing? If I understood correctly there is only a day or two left to enter the final 7.6 miniAOD re-production.

**Comment by slava77** [comment]
it's more in the "orp-pending" stage now (general release building decision)

**Comment by davidlange6** [comment]
just to clarify "The formal procedure is to merge new features in an open release cycle first and then backport to earlier release cycles and this requirement has been met now." is not correct. a validation is needed (here we are being asked/agreed to skip it..)

**Comment by ferencek** [comment]
@davidlange6, thank you for the clarification.

**Pull request merged**